OpenSubtitlesHandler/Util.cs

```diff
@@ -0,0 +1,152 @@
+        public static Action<string> OnHttpUpdate = _ => {};
+        private static string _version = string.Empty;
+
+        public static DateTime NextReset
+        {
+            get
+            {
+                // download limits get reset every day at midnight (UTC)
+                var now = DateTime.UtcNow;
+
+                return new DateTime(now.Year, now.Month, now.Day).AddDays(1).AddMinutes(1);
+            }
+        }
+
+        internal static void SetVersion(string version)
+        {
+            Util._version = version;
+        }
+
+        /// <summary>
+        /// Compute movie hash
+        /// </summary>
+        /// <returns>The hash as Hexadecimal string</returns>
+        public static string ComputeHash(Stream stream)
+        {
+            var hash = ComputeMovieHash(stream);
+            return Convert.ToHexString(hash).ToLowerInvariant();
+        }
+
+        /// <summary>
+        /// Serialize object into JSON
+        /// </summary>
+        /// <returns>JSON string of the object</returns>
+        public static string Serialize(object o)
+        {
+            return JsonSerializer.Serialize(o);
+        }
+
+        /// <summary>
+        /// Deserialize object from JSON
+        /// </summary>
+        /// <returns>Deserialized object</returns>
+        public static T Deserialize<T>(string str)
+        {
+            return JsonSerializer.Deserialize<T>(str, SerializerOpts);
+        }
+
+        /// <summary>
+        /// Compute hash of specified movie stream
+        /// </summary>
+        /// <returns>Hash of the movie</returns>
+        private static byte[] ComputeMovieHash(Stream input)
+        {
+            using (input)
+            {
+                var streamSize = input.Length;
+                var lHash = streamSize;
+
+                long i = 0;
+                byte[] buffer = new byte[sizeof(long)];
+
+                while (i < 65536 / sizeof(long) && (input.Read(buffer, 0, sizeof(long)) > 0))
+                {
+                    i++;
+                    lHash += BitConverter.ToInt64(buffer, 0);
+                }
+
+                input.Position = Math.Max(0, streamSize - 65536);
+                i = 0;
+
+                while (i < 65536 / sizeof(long) && (input.Read(buffer, 0, sizeof(long)) > 0))
+                {
+                    i++;
+                    lHash += BitConverter.ToInt64(buffer, 0);
+                }
+
+                byte[] result = BitConverter.GetBytes(lHash);
+                Array.Reverse(result);
+
+                return result;
+            }
+        }
+
+        internal static async Task<(string, Dictionary<string, string>, HttpStatusCode)> SendRequestAsync(string url, HttpMethod method, object body, Dictionary<string, string> headers, CancellationToken cancellationToken)
+        {
+            if (!HttpClient.DefaultRequestHeaders.Contains("User-Agent"))
```

Comment:
Use `HeaderNames.UserAgent` from the `Microsoft.Net.Http.Headers` namespace.

Reply:
Done in 65763b6e4766b8c5c8362bd63cd94698a313c6bb